File: minimum-increment-to-make-array-unique.cpp
class Solution {
public:
    int minIncrementForUnique(std::vector<int>& nums) {
        int n = nums.size();
        int res = 0;

        std::sort(nums.begin(), nums.end());

        std::multiset<int> absent;
        for (int i = nums[0]; i < nums[n - 1] + n; i++) {
            absent.insert(i);
        }

        for (int i = 0; i < n; i++) {
            auto it = absent.lower_bound(nums[i]);
            int currNum = *it;
            res += currNum - nums[i];
            absent.erase(it);
        }

        return res;
    }
};

File: minimum-increment-to-make-array-unique.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        nums.sort()
        for i in range(1, n):
            if nums[i] <= nums[i - 1]:
                res += nums[i - 1] + 1 - nums[i]
                nums[i] = nums[i - 1] + 1
        return res